### PR TITLE
🌱 bump and sync go dependencies

### PR DESCRIPTION
**Pull request by tuminoid**
Bump key go dependencies, and make the versions same across go.mod files.

### CODE REVIEW

File: test/go.mod [hunk 1]
```diff
@@ -9,19 +9,31 @@ require (
...
+	github.com/felixge/httpsnoop v1.0.4 // indirect
+	github.com/go-logr/stdr v1.2.2 // indirect
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.46.0 // indirect
+	go.opentelemetry.io/otel v1.28.0 // indirect
+	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp v1.28.0 // indirect
+	go.opentelemetry.io/otel/metric v1.28.0 // indirect
+	go.opentelemetry.io/otel/sdk v1.28.0 // indirect
+	go.opentelemetry.io/otel/trace v1.28.0 // indirect
+)
+
```
Comment (adilGhaffarDev) [inline test/go.mod:36]: can we put them in the same `require` block? line 25 and line 37.

Comment (tuminoid) [reply test/go.mod]: Removed docker/docker bump which caused that issue and also caused linters to explode.

**Review by kashifest — COMMENTED** [review]
/approve

**Pull request merged**